examples/step-89/step-89.cc

```diff
@@ -0,0 +1,1884 @@
+      }
+
+    // Quadrature points are arbitrarily distributed on each non-matching
+    // face. Therefore, we have to make use of FEPointEvaluation.
+    // FEPointEvaluation needs NonMatching::MappingInfo to work at the correct
+    // quadrature points that are in sync with used FERemoteEvaluation object.
+    // In the case of mortaring, we have to use the weights provided by the
+    // quadrature rules that are used to setup NonMatching::MappingInfo.
+    // Therefore we have to set the flag use_global_weights.
+    typename NonMatching::MappingInfo<dim, dim, Number>::AdditionalData
+      additional_data;
+    additional_data.use_global_weights = true;
+
+    // Setup NonMatching::MappingInfo with needed update flags and
+    // additional_data.
+    auto nm_mapping_info =
+      std::make_shared<NonMatching::MappingInfo<dim, dim, Number>>(
+        mapping,
+        update_values | update_JxW_values | update_normal_vectors |
+          update_quadrature_points,
+        additional_data);
+
+    // Reinit faces with the same vector of quadratures that is used to setup
+    // the remote communicator.
+    nm_mapping_info->reinit_faces(
+      matrix_free.get_dof_handler().get_triangulation().active_cell_iterators(),
+      global_quadrature_vector);
+
+    // Setup inverse mass operator.
+    const auto inverse_mass_operator =
+      std::make_shared<InverseMassOperator<dim, Number>>(matrix_free);
+
+    // Setup the acoustic operator. Using this constructor makes the
+    // operator use Nitsche-type mortaring.
+    const auto acoustic_operator =
+      std::make_shared<AcousticOperator<dim, Number, remote_value_type>>(
+        matrix_free,
+        material_data,
+        non_matching_faces,
+        pressure_r,
+        velocity_r,
+        c_r,
+        rho_r,
+        nm_mapping_info);
+
+    // Compute the the maximum speed of sound, needed for the computation of
+    // the time-step size.
+    double speed_of_sound_max = 0.0;
+    for (const auto &mat : materials)
+      speed_of_sound_max = std::max(speed_of_sound_max, mat.second.first);
+
+    ConditionalOStream pcout(
+      std::cout, (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0));
+
+    // Setup time integrator.
+    RungeKutta2<dim, Number, remote_value_type> time_integrator(
+      inverse_mass_operator, acoustic_operator);
+
+    // Run time loop with Courant number 0.1.
+    time_integrator.run(matrix_free,
+                        /*Cr*/ 0.2,
+                        end_time,
+                        speed_of_sound_max,
+                        initial_condition,
+                        vtk_prefix);
+  }
+} // namespace Step89
+
+
+// @sect3{Driver}
+//
+// Finally, the driver executes the different versions of handling non-matching
+// interfaces.
+int main(int argc, char *argv[])
+{
+  using namespace dealii;
+  constexpr int dim = 2;
+  using Number      = double;
+
+  Utilities::MPI::MPI_InitFinalize mpi(argc, argv, 1);
```

Comment:
I guess the code also works with threading enabled!?

```suggestion
  Utilities::MPI::MPI_InitFinalize mpi(argc, argv);
```